src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -192,7 +253,18 @@ internal static class MethodGenerator
 
                     if (method.IsStatic)
                     {
-                        tokens.Add(SyntaxFactory.Token(SyntaxKind.StaticKeyword));
+                        if (extractLocalFunction)
+                        {
+                            var preferStaticLocalFunction = workspace.Options.GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction).Value;
+                            if (preferStaticLocalFunction)
+                            {
+                                tokens.Add(SyntaxFactory.Token(SyntaxKind.StaticKeyword));
```

Comment:
you also need to check the language version to make sure static local functions are legal.  please add test for this with relevant language versions.